shell_integration/MacOSX/CMakeLists.txt

```diff
@@ -7,6 +7,9 @@ add_custom_target( legacy_mac_overlayplugin ALL
         OC_SOCKETAPI_TEAM_IDENTIFIER_PREFIX=${SOCKETAPI_TEAM_IDENTIFIER_PREFIX}
     COMMENT building Legacy Mac Overlay icons)
 
+# Contrary to popular belief, this is called like this no matter what theme/OEM.
+set(OC_OEM_SHARE_ICNS "${CMAKE_BINARY_DIR}/src/gui/ownCloud.icns")
```

Comment:
I added
```add_dependencies(mac_overlayplugin ${APPLICATION_EXECUTABLE}) # for the ownCloud.icns to be generated```

Then it works for me with ```-j5``` .. do you want to retry? (make sure to clean and remove the ownCloud.icns)